activesupport/lib/active_support/core_ext/numeric/conversions.rb

```diff
@@ -134,6 +134,12 @@ def to_formatted_s(*args)
   deprecate to_formatted_s: :to_s
 end
 
-[Fixnum, Bignum, Float, BigDecimal].each do |klass|
-  klass.prepend(ActiveSupport::NumericWithFormat)
+# Ruby 2.4+ unifies Fixnum & Bignum into Integer.
+if Integer == Fixnum
```

Comment:
Fair enough. If nothing else we could move the `Integer.prepend` case outside of the conditional. Perhaps something like this:

```ruby
Integer.prepend(ActiveSupport::NumericWithFormat)

if defined?(Bignum) && Bignum.ancestors.exclude?(ActiveSupport::NumericWithFormat)
  Bignum.prepend(ActiveSupport::NumericWithFormat)
end
```